### PR TITLE
Neverprompt

### PR DESCRIPTION
**Pull request by plafrance**
I Added «Always» and «Never» as possible answers to a prompt. I used «Never» in the password saving prompt. Also, to achieve the same result for browser features, I added a database named features.db on the model from bookmark.db and autofill.db. We can now always or never allow those features to be on.

Also, I slightly improved the «smooth» scrolling by allowing set_scroll_pos and scroll_by to perform a smooth scroll or not. Smooth scrolling being kind of slow on occasion, it's not always desirable.

### CODE REVIEW

File: webmacs/webbuffer.py [hunk 1]
```diff
@@ -175,11 +175,13 @@ def set_keymap_mode(self, enabled):
     def async_scroll_pos(self, func):
         self.runJavaScript("[window.pageXOffset, window.pageYOffset]", func)
 
-    def set_scroll_pos(self, x=0, y=0):
```
Comment (parkouss) [inline webmacs/webbuffer.py]: please remove that code from the PR

**Comment by parkouss** [comment]
Hey @plafrance , it seems to be working well!

If you want, I can take care of removing the parts I don't want to see in that pr (about the smooth scrolling), fix the neats I saw and merge everything. Or you can take care of that, tell me how you want to proceed.

In any case, this is a quite awesome PR. Now we will have to to create some web pages to be able to see and change what is stored in those databases! Anyway, really good stuff, thanks for the work here.

**Comment by plafrance** [comment]
Hi Julien,
I'll take care of it. I'm sorry my command of git/github is not great; i'm still learning. I guess I commited the smoothing part on the wrong branch and didn't know how to correct my mistake. I'll be fixing it today.

**Comment by plafrance** [comment]
Fixed. If you have an idea how to avoid the import of FormData, go ahead; I can't think of a better way.

File: webmacs/autofill/prompt.py [hunk 2]
```diff
@@ -19,15 +19,25 @@
...
-        YesNoPrompt.__init__(self, "Save password ?", buffer)
+        YesNoPrompt.__init__(self, "Save password ?", buffer, never=True)
         self.buffer = buffer
         self.autofill = autofill
         self.formdata = formdata
         self.closed.connect(self.__on_closed)
 
     def __on_closed(self):
+        # late import to avoid cyclic dependencies
+        from . import FormData
```
[the remark this answers is not in the document]
Comment (parkouss) [reply webmacs/autofill/prompt.py]: oh, that make sense, thanks. I am thinking about adding a copy method inside the FormData class maybe, I'll look into it.

**Pull request merged**